refaster-runner/src/main/java/tech/picnic/errorprone/refaster/runner/RefasterRuleSelector.java

```diff
@@ -0,0 +1,554 @@
+      // XXX: Also include the package name if not `java.lang`; it must be present.
+      if (RefasterIntrospection.isUClassIdent(node)) {
+        for (Set<String> ids : identifierCombinations) {
+          ids.add(getSimpleName(RefasterIntrospection.getTopLevelClass(node)));
+          ids.add(getIdentifier(node));
+        }
+      } else if (node instanceof UStaticIdent) {
+        IdentifierTree subNode = RefasterIntrospection.getClassIdent((UStaticIdent) node);
+        for (Set<String> ids : identifierCombinations) {
+          ids.add(getSimpleName(RefasterIntrospection.getTopLevelClass(subNode)));
+          ids.add(getIdentifier(subNode));
+          ids.add(node.getName().toString());
+        }
+      }
+
+      return null;
+    }
+
+    private static String getIdentifier(IdentifierTree tree) {
+      return getSimpleName(tree.getName().toString());
+    }
+
+    private static String getSimpleName(String fqcn) {
+      int index = fqcn.lastIndexOf('.');
+      return index < 0 ? fqcn : fqcn.substring(index + 1);
+    }
+
+    @Nullable
+    @Override
+    public Void visitMemberReference(
+        MemberReferenceTree node, List<Set<String>> identifierCombinations) {
+      super.visitMemberReference(node, identifierCombinations);
+      String id = node.getName().toString();
+      identifierCombinations.forEach(ids -> ids.add(id));
+      return null;
+    }
+
+    @Nullable
+    @Override
+    public Void visitMemberSelect(MemberSelectTree node, List<Set<String>> identifierCombinations) {
+      super.visitMemberSelect(node, identifierCombinations);
+      String id = node.getIdentifier().toString();
+      identifierCombinations.forEach(ids -> ids.add(id));
+      return null;
+    }
+
+    @Nullable
+    @Override
+    public Void visitAssignment(AssignmentTree node, List<Set<String>> identifierCombinations) {
+      registerOperator(node, identifierCombinations);
+      return super.visitAssignment(node, identifierCombinations);
+    }
+
+    @Nullable
+    @Override
+    public Void visitCompoundAssignment(
+        CompoundAssignmentTree node, List<Set<String>> identifierCombinations) {
+      registerOperator(node, identifierCombinations);
+      return super.visitCompoundAssignment(node, identifierCombinations);
+    }
+
+    @Nullable
+    @Override
+    public Void visitUnary(UnaryTree node, List<Set<String>> identifierCombinations) {
+      registerOperator(node, identifierCombinations);
+      return super.visitUnary(node, identifierCombinations);
+    }
+
+    @Nullable
+    @Override
+    public Void visitBinary(BinaryTree node, List<Set<String>> identifierCombinations) {
+      registerOperator(node, identifierCombinations);
+      return super.visitBinary(node, identifierCombinations);
+    }
+
+    private static void registerOperator(
+        ExpressionTree node, List<Set<String>> identifierCombinations) {
+      identifierCombinations.forEach(ids -> ids.add(treeKindToString(node.getKind())));
```

Comment:
The identifier does not change.
```suggestion
      String id = treeKindToString(node.getKind());
      identifierCombinations.forEach(ids -> ids.add(id));
```